xblocks_contrib/annotatable/annotatable.py

```diff
@@ -0,0 +1,115 @@
+
+import pkg_resources
+from django.utils import translation
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Integer, Scope
+from xblockutils.resources import ResourceLoader
+
+
+class AnnotatableXBlock(XBlock):
+    """
+    TO-DO: document what your XBlock does.
+    """
+
+    # Fields are defined on the class.  You can access them in your code as
+    # self.<fieldname>.
+
+    # TO-DO: delete count, and define your own fields.
+    count = Integer(
+        default=0,
+        scope=Scope.user_state,
+        help="A simple counter, to show something happening",
+    )
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    # TO-DO: change this view to display your data your own way.
+    def student_view(self, context=None):
+        """
+        Create primary view of the AnnotatableXBlock, shown to students when viewing courses.
+        """
+        if context:
+            pass  # TO-DO: do something based on the context.
+        html = self.resource_string("static/html/annotatable.html")
+        frag = Fragment(html.format(self=self))
+        frag.add_css(self.resource_string("static/css/annotatable.css"))
+
+        # Add i18n js
+        statici18n_js_url = self._get_statici18n_js_url()
+        if statici18n_js_url:
+            frag.add_javascript_url(
+                self.runtime.local_resource_url(self, statici18n_js_url)
+            )
+
+        frag.add_javascript(self.resource_string("static/js/src/annotatable.js"))
+        frag.initialize_js("AnnotatableXBlock")
+        return frag
+
+    # TO-DO: change this handler to perform your own actions.  You may need more
+    # than one handler, or you may not need any handlers at all.
+    @XBlock.json_handler
+    def increment_count(self, data, suffix=""):
+        """
+        Increments data. An example handler.
+        """
+        if suffix:
+            pass  # TO-DO: Use the suffix when storing data.
+        # Just to show data coming in...
+        assert data["hello"] == "world"
+
+        self.count += 1
+        return {"count": self.count}
+
+    # TO-DO: change this to create the scenarios you'd like to see in the
+    # workbench while developing your XBlock.
+    @staticmethod
+    def workbench_scenarios():
+        """Create canned scenario for display in the workbench."""
+        return [
+            (
+                "AnnotatableXBlock",
+                """<annotatable/>
+             """,
+            ),
+            (
+                "Multiple AnnotatableXBlock",
+                """<vertical_demo>
+                <annotatable/>
+                <annotatable/>
+                <annotatable/>
+                </vertical_demo>
+             """,
+            ),
+        ]
+
+    @staticmethod
+    def _get_statici18n_js_url():
```

Comment:
We need to study the latest way of supporting `i18n`
as in DoneXBlock it implemented in this way
https://github.com/openedx/DoneXBlock/blob/master/done/done.py#L33